crates/egui/src/context.rs

```diff
@@ -2374,6 +2374,17 @@ impl Context {
         self.memory_mut(|mem| mem.areas_mut().move_to_top(layer_id));
     }
 
+    /// Mark the `child` layer as a sublayer of `parent`.
+    ///
+    /// Sublayers are moved directly above the parent layer at the end of the frame. This is mainly
+    /// intended for adding a new [`Area`] inside a [`Window`].
+    ///
+    /// This currently only supports one level of nesting. If `parent` is a sublayer of another
+    /// layer, the behavior is unspecified.
+    pub fn set_sublayer(&self, parent: LayerId, child: LayerId) {
```

Comment:
Maybe this could be called something like `set_parent_window` to be clearer? Technically the parent doesn't have to be a Window, but that is the intended use case, and a more complicated hierarchy would not work well with the current reordering algorithm. I'm not really sure what to call it.

Reply:
I like the current name, but we should document that this is a sticky setting, and therefor there should also be some way to clear it. Or we make it an immediate-mode function instead and clear it at the end of each frame

Reply:
I actually meant for it to be immediate, like the wants_to_be_on_top set. I'm going to fix that.